### PR TITLE
Adding tests for buck lint.

### PR DESCRIPTION
**Pull request by clonetwin26**
This adds integration and unit tests for the buck lint useage.

https://github.com/uber/okbuck/issues/340

### CODE REVIEW

File: libraries/customLintLibrary/build.gradle [hunk 1]
```diff
@@ -0,0 +1,27 @@
+apply plugin: 'java'
+apply plugin: 'application'
+
+sourceCompatibility = 1.7
```
Comment (kageiit) [inline libraries/customLintLibrary/build.gradle]: source and repos are configured at root. they can be removed

File: libraries/customLintLibrary/src/main/java/com/uber/lint/SystemCurrentTimeMillisDetector.java [hunk 2]
```diff
@@ -0,0 +1,63 @@
...
+public class SystemCurrentTimeMillisDetector extends Detector implements Detector.JavaScanner {
+
+    public static final String CHECK_METHOD_TO_EXCLUDE = "currentTimeMillis";
+    public static final String CHECK_PACKAGE_TO_EXCLUDE = "java.lang.System";
+
+    public static final String ISSUE_ID = "DontUseSystemTime";
+    public static final Issue ISSUE = Issue.create(
+            ISSUE_ID,
+            "Don't use System.currentTimeMillis()",
+            "This method is blocked from useage. Please prefer a Clock, as this is "
```
Comment (kageiit) [inline libraries/customLintLibrary/src/main/java/com/uber/lint/SystemCurrentTimeMillisDetector.java]: Remove references to `Clock`

File: libraries/customLintLibrary/src/main/resources/META-INF/MANIFEST.MF [hunk 3]
```diff
@@ -0,0 +1,3 @@
+Manifest-Version: 1.0
+Lint-Registry: com.uber.lint.LintRegistry
+Created-By: 1.8.0_102 (Oracle Corporation)
```
Comment (kageiit) [inline libraries/customLintLibrary/src/main/resources/META-INF/MANIFEST.MF]: Don't need the last line

File: libraries/emptylibrary/src/test/java/com/uber/okbuck/example/empty/BuckLintUnitTest.java [hunk 4]
```diff
@@ -0,0 +1,38 @@
+package com.uber.okbuck.example.empty;
```
Comment (kageiit) [inline libraries/emptylibrary/src/test/java/com/uber/okbuck/example/empty/BuckLintUnitTest.java]: Remove this file

File: libraries/emptylibrary/src/test/java/com/uber/okbuck/example/empty/EmptyLibraryUnitTest.java [hunk 5]
```diff
@@ -25,4 +28,13 @@ public void resource_loading() {
         Context context = (Context) RuntimeEnvironment.application;
         assertEquals(context.getResources().getString(R.string.empty_release_string), "empty_release_string");
     }
+
```
Comment (kageiit) [inline libraries/emptylibrary/src/test/java/com/uber/okbuck/example/empty/EmptyLibraryUnitTest.java]: Remove this test

File: libraries/lintErrorLibrary/build.gradle [hunk 6]
```diff
@@ -0,0 +1,16 @@
+apply plugin: 'com.android.library'
+apply plugin: 'me.tatarka.retrolambda'
+
+android {
+    lintOptions {
+        disable 'HardcodedDebugMode'
```
Comment (kageiit) [inline libraries/lintErrorLibrary/build.gradle]: Move it to the xml instead

File: libraries/lintErrorLibrary/src/main/res/values/strings.xml [hunk 7]
```diff
@@ -0,0 +1,4 @@
+<?xml version="1.0" encoding="utf-8"?>
+<resources>
+    <string name="empty_release_string">empty_release_string</string>
```
Comment (kageiit) [inline libraries/lintErrorLibrary/src/main/res/values/strings.xml]: Remove this file

File: libraries/lintErrorLibrary/src/test/java/com/uber/okbuck/example/empty/BuckLintUnitTest.java [hunk 8]
```diff
@@ -0,0 +1,38 @@
+package com.uber.okbuck.example.empty;
```
Comment (kageiit) [inline libraries/lintErrorLibrary/src/test/java/com/uber/okbuck/example/empty/BuckLintUnitTest.java]: Remove this file

File: libraries/lintErrorLibrary/src/test/java/com/uber/okbuck/example/empty/LintErrorLibraryTest.java [hunk 9]
```diff
@@ -0,0 +1,26 @@
+package com.uber.okbuck.example.empty;
```
Comment (kageiit) [inline libraries/lintErrorLibrary/src/test/java/com/uber/okbuck/example/empty/LintErrorLibraryTest.java]: Remove this file

**Comment by clonetwin26** [comment]
When you get the chance this should be ready to pull. 

Battle tested the conditional:

if grep -q 'id=\"DontUseSystemTime\"' 'buck-out/gen/libraries/lintErrorLibrary/lint_debug/lint_debug_out/lint-results.xml'; then

by switching around the DontUseSystemTime string, and it works on ci as expected.

**Pull request merged**